### PR TITLE
Add FeatureGates for NamespacedServiceBrokers and CatalogRestrictions

### PR DESCRIPTION
**Pull request by jboyd01**

### CODE REVIEW

File: roles/openshift_service_catalog/defaults/main.yml [hunk 1]
```diff
@@ -1,6 +1,11 @@
 ---
 openshift_service_catalog_remove: false
+
+# Feature Gates
 openshift_service_catalog_async_bindings_enabled: true
+openshift_service_catalog_namespaced_service_brokers_enabled: true
+openshift_service_catalog_catalog_restrictions_enabled: true
```
Comment (jpeeler) [inline roles/openshift_service_catalog/defaults/main.yml]: Did you intend for the name to be "...catalog_catalog..."?

File: roles/openshift_service_catalog/templates/controller_manager.j2 [hunk 2]
```diff
@@ -46,6 +46,14 @@ spec:
 {% if openshift_service_catalog_async_bindings_enabled | bool %}
         - --feature-gates
         - AsyncBindingOperations=true
+{% endif %}
+{% if openshift_service_catalog_namespaced_service_brokers_enabled | bool %}
+        - --feature-gates
+        - NamespacedServiceBroker=true
+{% endif %}
+{% if openshift_service_catalog_catalog_restrictions_enabled | bool %}
```
Comment (pmorie) [inline roles/openshift_service_catalog/templates/controller_manager.j2]: I think it's fine to just have this on by default. I don't think it should be behind a feature gate.

File: roles/openshift_service_catalog/defaults/main.yml (hunk 1 above)
Comment (jboyd01) [reply roles/openshift_service_catalog/defaults/main.yml]: Yes.  The feature is CatalogRestrictions.

**Comment by pmorie** [comment]
I'll delegate lgtm to @jpeeler but LGTM

**Comment by sdodson** [comment]
Ansible code lgtm, most recent commit removed the ability to disable CatalogRestrictions where as in original commit it was enabled by default with the option to disable it. Not sure if that's intentional.
/assign @jpeeler

**Comment by jboyd01** [comment]
Thanks @sdodson, yes, intentional change per Paul's review.

**Comment by sdodson** [comment]
Fedora tests are broken as of late, they should be fixed or removed by end of day today. Feel free to reach out to me to get anything merged that's only blocked on them.

**Pull request merged**